reactor-netty-http/src/test/java/reactor/netty/http/client/ConnectionPoolTests.java

```diff
@@ -0,0 +1,353 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.netty.BaseHttpTest;
+import reactor.netty.DisposableServer;
+import reactor.netty.channel.ChannelMetricsRecorder;
+import reactor.netty.http.HttpProtocol;
+import reactor.netty.http.server.HttpServer;
+import reactor.netty.resources.ConnectionProvider;
+import reactor.netty.resources.LoopResources;
+import reactor.util.function.Tuple2;
+
+import java.security.cert.CertificateException;
+import java.time.Duration;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * This class tests https://github.com/reactor/reactor-netty/issues/1472
+ */
+class ConnectionPoolTests extends BaseHttpTest {
+
+	static DisposableServer server1;
+	static DisposableServer server2;
+	static DisposableServer server3;
+	static DisposableServer server4;
+	static ConnectionProvider provider;
+	static LoopResources loop;
+	static ChannelMetricsRecorder metricsRecorder;
+
+	HttpClient client;
+
+	@BeforeAll
+	static void prepare() throws CertificateException {
+		HttpServer server = createServer();
+
+		server1 = server.handle((req, res) -> res.sendString(Mono.just("server1-ConnectionPoolTests")))
+		                .bindNow();
+
+		server2 = server.handle((req, res) -> res.sendString(Mono.just("server2-ConnectionPoolTests")))
+		                .bindNow();
+
+		SelfSignedCertificate cert = new SelfSignedCertificate();
+		SslContextBuilder sslContextBuilder = SslContextBuilder.forServer(cert.certificate(), cert.privateKey());
+
+		server3 = server.protocol(HttpProtocol.H2, HttpProtocol.HTTP11)
+		                .secure(spec -> spec.sslContext(sslContextBuilder))
+		                .handle((req, res) -> res.sendString(Mono.just("server3-ConnectionPoolTests")))
+		                .bindNow();
+
+		server4 = server.secure(spec -> spec.sslContext(sslContextBuilder))
+		                .handle((req, res) -> res.sendString(Mono.just("server4-ConnectionPoolTests")))
+		                .bindNow();
+
+		provider = ConnectionProvider.create("ConnectionPoolTests", 1);
+
+		loop = LoopResources.create("ConnectionPoolTests");
+
+		metricsRecorder = Mockito.mock(ChannelMetricsRecorder.class);
+	}
+
+	@AfterAll
+	static void tearDown() {
+		server1.disposeNow();
+		server2.disposeNow();
+		server3.disposeNow();
+		server4.disposeNow();
+		provider.disposeLater()
+		        .block(Duration.ofSeconds(5));
+		loop.disposeLater()
+		    .block(Duration.ofSeconds(5));
+	}
+
+	@BeforeEach
+	void setUp() {
+		client = HttpClient.create(provider);
+	}
+
+	@Test
+	void testClientWithPort() {
+		HttpClient localClient = client.port(server1.port());
+		checkExpectations(
+				"server1-ConnectionPoolTests",
+				"server2-ConnectionPoolTests",
+				localClient,
+				localClient.port(server2.port()));
```

Comment:
just a readability comment: why not put the initialisation of second `HttpClient` in its own line as `localClient2` next to `localClient`, rename the other one to `localClient1` and pass both variables to the `checkExpectations` method? this would simplify the call and allow to quickly get a sense of the two http calls made by each test.

Reply:
I fixed that PTAL